### ISSUE TITLE
lua Track: 'busted' command not found.

### ISSUE THREAD

**Issue by aule555** [exercism/exercism#5032]
Hi everyone, i'm not easy with Lua but i've tried to run this command in my terminal like it's simply described in the readme exercice. I've searched for an anwser but i've found anything. Someone have an idea ?
Regards.

**Comment by glennj** [comment]
Check the INSTALLATION document for the Lua track: https://github.com/exercism/lua/blob/master/docs/INSTALLATION.md

**Comment by rpottsoh** [comment]
What should be checked for?  What problem are you seeing?

**Comment by aule555** [comment]
@glennj i've tried to reinstall all for busted and i have this error during the installation:
```

busted 2.0.0-1 depends on luafilesystem >= 1.5.0 (not installed)
Installing https://luarocks.org/luafilesystem-1.7.0-2.src.rock
gcc -O2 -fPIC -I/usr/include/lua5.2 -c src/lfs.c -o src/lfs.o
src/lfs.c:66:17: fatal error: lua.h: Aucun fichier ou dossier de ce type
 #include <lua.h>
                 ^
compilation terminated.

Error: Failed installing dependency: https://luarocks.org/luafilesystem-1.7.0-2.src.rock - Build error: Failed compiling object src/lfs.o
```

@rpottsoh i'm on the second exercice from the lua track, "Hamming", and i just want test my exercice before to send it. The README just say "To run the tests, run the command `busted` from within the exercise directory." But it seems like there no busted command, even in the exercice folder.

**Comment by NobbZ** [comment]
> i've tried to reinstall all for busted and i have this error during the installation:

So then install its dependencies. From the error message it seems as if it were missing the `lua` development headers. Since you haven't told us which operating system you are using, we can't really tell you the name of the package or from where you could install it.

For ubuntu based systems its probably `liblua5.3-dev` you are missing, thats explicitely mentioned on the install page linked.

If you are on a different linux please tell us and we'll help you find the correct package.

**Comment by aule555** [comment]
Nevermind, i'm on Bunsenlabs (debian 9 - based)

**Comment by NobbZ** [comment]
I am not used to debian based systems, but as far as I remember, they have similar naming conventions as ubuntu.

So `libluaX.Y-dev` (where `X.Y` is the version of the `lua` you installed) should work.

**Comment by aule555** [comment]
I have already installed lua5.3 and it's apparently the appropriate version. I've got the same problem.

**Comment by aule555** [comment]
yes i've done it and i have this error during busted installation.

> @glennj i've tried to reinstall all for busted and i have this error during the installation:
> 
> ```
> 
> busted 2.0.0-1 depends on luafilesystem >= 1.5.0 (not installed)
> Installing https://luarocks.org/luafilesystem-1.7.0-2.src.rock
> gcc -O2 -fPIC -I/usr/include/lua5.2 -c src/lfs.c -o src/lfs.o
> src/lfs.c:66:17: fatal error: lua.h: Aucun fichier ou dossier de ce type
>  #include <lua.h>
>                  ^
> compilation terminated.
> 
> Error: Failed installing dependency: https://luarocks.org/luafilesystem-1.7.0-2.src.rock - Build error: Failed compiling object src/lfs.o
> ```
> 
> @rpottsoh i'm on the second exercice from the lua track, "Hamming", and i just want test my exercice before to send it. The README just say "To run the tests, run the command `busted` from within the exercise directory." But it seems like there is no busted command, even in the exercice folder.

**Comment by glennj** [comment]
OK, looks like the header files aren't getting installed. What happens when you do this?

    $ sudo apt-get install liblua5.3-dev

**Comment by aule555** [comment]
```
$ sudo apt-get install liblua5.3-dev

Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances       
Lecture des informations d'état... Fait
liblua5.3-dev is already the newest version (5.3.3-1).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 1 non mis à jour.
```
I know there is french but i see any error message. Should i try to uninstall / re-install ?

**Comment by glennj** [comment]
I notice that the error message for building luafilesystem includes this:

    gcc -O2 -fPIC -I/usr/include/lua5.2 -c src/lfs.c -o src/lfs.o

So I wonder why it found `/usr/includer/lua5.2` when you're talking about having installed lua5.3

**Comment by aule555** [comment]
should i try an `apt-get autoremove lua lua5.3-dev`  ?

**Comment by glennj** [comment]
I'd try to figure out how the luafilesystem build decided lua5.2  should be in the include path.

Also, what's in `$HOME/.luarocks`? What's in `/usr/local/etc/luarocks`? `/usr/local/lib/luarocks`?

**Comment by aule555** [comment]
```
$ ls $HOME/.luarocks
ls: impossible d'accéder à '/home/c1c3r0/.luarocks': Aucun fichier ou dossier de ce type

```
anything found.

```
$ ls /usr/local/etc/luarocks
ls: impossible d'accéder à '/usr/local/etc/luarocks': Aucun fichier ou dossier de ce type

```
same.

```
$ ls /usr/local/lib/luarocks
rocks
$ ls -a /usr/local/lib/luarocks/rocks
.  ..  index.html  lua_cliargs  manifest  manifest-5.1  manifest-5.2  manifest-5.3
```
```
$ type -a luarocks
luarocks est /usr/bin/luarocks
```

**Comment by NobbZ** [comment]
Can you please prefix all commands with `LANG=C` to see their unlocalized output? Its hard for us to follow the french output…

**Comment by aule555** [comment]
Nevermind.. i've edited my previous message !

**Comment by sshine** [comment]
@aule555: Have you found a solution to your problem?

**Comment by aule555** [comment]
Unfortunately no, i'm still searching.

**Comment by NobbZ** [comment]
Please remove all `lua` related packages from you system and then install only the 5.3 packages.

**Comment by aule555** [comment]
Hi, i have noticed 2 things.
During the installation, i have this warning:
`$- sudo apt install luarocks
...
Making manifest for /usr/local/lib/luarocks/rocks
Warning: This looks like a local rocks tree, but you did not pass --local-tree.
Generating index.html for /usr/local/lib/luarocks/rocks
...
`
Futhermore:
`CONFIGURATION
	Lua version: 5.3
	Configuration files:
		System: /etc/luarocks/config.lua (ok)
		User  : /home/$USER/.luarocks/config-5.3.lua (not found)

	Rocks trees in use: 
		/home/c1c3r0/.luarocks
		/usr/local
`
Config file in user is not found. Peharps these can be a hint ?
